Update remote-services.md (#2630)
cap-js/docs: java/cqn-services/remote-services.md

[Learn more about consuming APIs from other IAS-Applications in the **SAP Cloud Identity Services documentation**.](https://help.sap.com/docs/cloud-identity-services/cloud-identity-services/consume-apis-from-other-applications){.learn-more}

> ##### Configuring the Authentication Strategy {#ias-destination-authentication-strategy}

Reviewer: I guess this is not needed.
```suggestion
##### Configuring the Authentication Strategy
```

> By default, when calling a remote IAS-based API through a destination, CAP propagates the current user's identity (`currentUser`). This means the remote API receives a token representing the currently logged-in user.

Reviewer 2: ```suggestion
By default, when calling a remote IAS-based API through a destination, CAP propagates the user identity associated with the current `RequestContext` (`currentUser`). If the user identity in the `RequestContext` is a named user, the names user will be propagated. If not, a technical user token will be requested.
```

Author: ```suggestion
By default, when calling a remote IAS-based API through a destination, CAP propagates the user identity associated with the current `RequestContext` (`currentUser`). If the user identity in the `RequestContext` is a named user, the named user will be propagated. If not, a technical user token will be requested.
```

> By default, when calling a remote IAS-based API through a destination, CAP propagates the user identity associated with the current `RequestContext` (`currentUser`). If the user identity in the `RequestContext` is a named user, the named user will be propagated. If not, a technical user token will be requested.

Reviewer: ```suggestion
By default, when calling a remote IAS-based API through a destination, CAP propagates the user identity associated with the current `RequestContext` (`currentUser`). If the `RequestContext` contains a named user, CAP propagates the named user. If not, CAP requests a technical user token.
```

> In some scenarios, especially for background processing or technical integrations, you may want to call the remote API as a technical user instead. The `destination.onBehalfOf` configuration allows you to control this behavior:

Reviewer 2: ```suggestion
In some scenarios, especially for background processing or technical integrations, you may want to enforce the call to the remote API as a technical user instead. The `destination.onBehalfOf` configuration allows you to control this behavior:
```

> In some scenarios, especially for background processing or technical integrations, you may want to enforce the call to the remote API as a technical user instead. The `destination.onBehalfOf` configuration allows you to control this behavior:

Reviewer: ```suggestion
For background processing or technical integrations, you may want to force the remote API call to use a technical user. Use the `destination.onBehalfOf` configuration to control this behavior:
```

::: code-group
```yaml [srv/src/main/resources/application.yaml]
cds:
  remote.services:
    RemoteIasService:
      destination:
        name: my-ias-destination
        onBehalfOf: systemUser
```

The following options are available:

| Value | Description |
|-------|-------------|
| `currentUser` | Propagates the named user if available, or falls back to a tenant-specific technical user. **(default)** |
| `systemUser` | Uses a tenant-specific technical user, based on the tenant set in the current Request Context. |
| `systemUserProvider` | Uses a technical user of the provider tenant. Useful for internal communication that does not require tenant-specific authorization. |

> ::: tip

Reviewer: ```suggestion
::: tip Behaves similar to binding-based configuration
```

> | `systemUserProvider` | Uses a technical user of the provider tenant. Useful for internal communication that is not authorized tenant-specifically. |

Reviewer: ```suggestion
| `systemUserProvider` | Uses a technical user of the provider tenant. Useful for internal communication that does not require tenant-specific authorization. |
```

This behaves identically to the [`onBehalfOf` option in binding-based configurations](#configuring-the-authentication-strategy). Use it when your IAS app-2-app communication is configured via a BTP destination with `cloudsdk.ias-dependency-name` rather than a direct service binding.
:::

> ::: warning

Reviewer: ```suggestion
::: warning Only applicable to IAS app-2-app destinations
```

> The `onBehalfOf` option only applies to IAS app-2-app destinations (destinations with the `cloudsdk.ias-dependency-name` property set). It has no effect on other destination types.

Reviewer: ```suggestion
The `onBehalfOf` option applies only to IAS app-2-app destinations (destinations with the `cloudsdk.ias-dependency-name` property set). It has no effect on other destination types.
```
